### PR TITLE
python310Packages.qbittorrent-api: init at 2023.7.52

### PR DESCRIPTION
**Pull request by savyajha**
## Description of changes

Adds the qbittorrent-api package, which is a python API client for qbittorrent.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
Please change PR and commit title to `python310packages.qbittorrent-api: init at 2023.7.52`

File: pkgs/development/python-modules/qbittorrent-api/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  propagatedBuildInputs = [
+    requests
+    six
+    urllib3
+    packaging
+  ];
+
+  buildInputs = [
+    setuptools
+  ];
```
Comment (natsukium) [inline pkgs/development/python-modules/qbittorrent-api/default.nix:30]:
```suggestion
  nativeBuildInputs = [
    setuptools
    wheel
  ];
```
We will have to add  `wheel` to `nativeBuildInputs`.
see https://github.com/NixOS/nixpkgs/pull/245509

File: pkgs/development/python-modules/qbittorrent-api/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "qbittorrentapi"
+  ];
+
+  meta = with lib; {
+    description = "Python client implementation for qBittorrent's Web API";
+    homepage = "https://github.com/rmartin16/qbittorrent-api";
+    changelog = "https://github.com/rmartin16/qbittorrent-api/blob/main/CHANGELOG.md";
```
Comment (natsukium) [inline pkgs/development/python-modules/qbittorrent-api/default.nix:42]:
```suggestion
    changelog = "https://github.com/rmartin16/qbittorrent-api/blob/v${version}/CHANGELOG.md";
```

**Comment by savyajha** [comment]
Made the requested changes.

File: pkgs/development/python-modules/qbittorrent-api/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  propagatedBuildInputs = [
+    requests
+    six
+    urllib3
+    packaging
+  ];
+
+  buildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/qbittorrent-api/default.nix:29]:
```suggestion
  nativeBuildInputs = [
```
I think we don't need these inputs in run-time.

Comment (savyajha) [reply pkgs/development/python-modules/qbittorrent-api/default.nix]: Made the change. Sorry, that was an oversight.

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Pull request merged**